src/database.ts

```diff
@@ -0,0 +1,15 @@
+import * as mongoose from "mongoose";
+
+class Database {
+  public db_connect(mongoUrl: string | undefined, mongoDb: string | undefined) {
+    mongoose.connect(`'${mongoUrl}/${mongoDb}'`,  { useNewUrlParser: true })
+      .then (() => {
+        console.log(`Database connection to ${mongoUrl}/${mongoDb} successful`);
+      })
+      .catch((err: any) => {
+        console.error(`Database connection error to ${mongoUrl}/${mongoDb}  ${err}`);
+      });
+  }
+}
+
+export const dbConnect = new Database().db_connect;
```

Comment:
Not sure about making the DB connection a singleton. Ideally we want to pool connections right? Not sure on best practise here, would be interested in getting your view

Reply:
I'm just reading up about pool connections and you are correct we would want to pool connections. I will look into changing the code for this.

Reply:
I don’t think it would matter if it is a singleton as it just sets the default connection as there should only be one default?

If we needed to make more connections for whatever reason, like to another database or the same database with different read/write settings we can create new connections using mongoose.createConnection()

https://mongoosejs.com/docs/connections.html#multiple_connections